ocp3config/ocp3_test.go

```diff
@@ -0,0 +1,55 @@
+package ocp3
+
+import (
+	"testing"
+
+	configv1 "github.com/openshift/api/legacyconfig/v1"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestConfigParseMaster(t *testing.T) {
+	testConfig := Config{
+		Masterf: "../examples/source/etc/origin/master/master-config.yaml",
+	}
+
+	expectedMasterConfig := configv1.MasterConfig{
+		AuthConfig: configv1.MasterAuthConfig{
+			RequestHeader: &configv1.RequestHeaderAuthenticationOptions{
+				ClientCA: "front-proxy-ca.crt",
+			},
+		},
+		EtcdClientInfo: configv1.EtcdConnectionInfo{
+			CA:   "master.etcd-ca.crt",
+			URLs: []string{"https://master-0.gildub2.lab.pnq2.cee.redhat.com:2379"},
+		},
+		OAuthConfig: &configv1.OAuthConfig{
+			MasterURL: "https://openshift.internal.gildub2.lab.pnq2.cee.redhat.com:443",
+			IdentityProviders: []configv1.IdentityProvider{
+				configv1.IdentityProvider{
+					Name: "htpasswd_auth",
+				},
+				configv1.IdentityProvider{
+					Name: "github123456789",
+				},
+			},
+		},
+	}
+
+	resMasterConfig := testConfig.ParseMaster()
+
```

Comment:
I decided to put some random fields values in out test object, assuming there is no need to check all of them.